### PR TITLE
TST: add message matches to pytest.raises in various tests GH30999

### PR DESCRIPTION
**Pull request by JoseNavy**
- [x] ref [#30999 ](https://github.com/pandas-dev/pandas/issues/30999)
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Files changed: 
pandas/tests/io/pytables/test_complex.py
pandas/tests/io/test_clipboard.py
pandas/tests/plotting/frame/test_frame_subplots.py

### CODE REVIEW

**Comment by JoseNavy** [comment]
I tried to follow the contributing guidelines as best as I could. If there's any problem, let me know.

File: pandas/tests/io/test_clipboard.py [hunk 1]
```diff
@@ -238,10 +238,11 @@ def test_read_clipboard_infer_excel(self, request, mock_clipboard):
         tm.assert_frame_equal(res, exp)
 
     def test_invalid_encoding(self, df):
+        msg = "invalid coding, encoding must be ascii"
         # test case for testing invalid encoding
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError, match=msg):
```
Comment (arw2019) [inline pandas/tests/io/test_clipboard.py:243]: looking at the CI the message being thrown is different

Comment (JoseNavy) [reply pandas/tests/io/test_clipboard.py:243]: I thought that the message didn't exist and wrote one, I'll fix them so they match with what's supposed to say. Where can I find the exact message?

Comment (arw2019) [reply pandas/tests/io/test_clipboard.py:243]: run the code that throws the error

**Comment by jreback** [comment]
thanks @JoseNavy

**Pull request merged**